pathd/path_pcep_controller.c

```diff
@@ -304,7 +304,7 @@ struct pcep_pcc_info *pcep_ctrl_get_pcc_info(struct frr_pthread *fpt,
 	struct ctrl_state *ctrl_state = get_ctrl_state(fpt);
 	struct pcep_pcc_info *args = XCALLOC(MTYPE_PCEP, sizeof(*args));
 	args->ctrl_state = ctrl_state;
-	strncpy(args->pce_name, pce_name, sizeof(args->pce_name));
+	strncpy(args->pce_name, pce_name, sizeof(args->pce_name)-1);
 	thread_execute(ctrl_state->self, pcep_thread_get_pcc_info_callback,
```

Comment:
I'm confused about what this is doing - it looks like it goes all the way through the lib/thread module infra in order to execute a callback and return? am I reading that right - if so, why?

Reply:
If you are talking about the `thread_execute` call, this is so the callback is executed "synchronously" in a different thread in a safe way without having to mess with mutex.

Reply:
wow - if you mean "in a different pthread", I don't think that's what's happening? what I see is that the caller ends up in `_thread_execute()` where a task struct gets allocated, holding the various args, and then that struct gets handed to `thread_call()` which ... calls the function. that's all in the context of the calling pthread, right? I know the use of the word "thread" is confusing sometimes, so I'm just trying to be super-clear to understand.

Reply:
ok, there may be something wrong there, but I want to clarify a bit. The idea is not really to run the code in a different pthread, it is to protect the callback as a mutex would do using FRR infrastructure. The other pthread is a FRR initialised pthread that is used only with the FRR concept of threads (events, timers  and such). This other pthread (`ctrl_state->self`) own some data structures that should only be modified in the context of this pthread or protected in some ways. The idea is that by calling `thread_execute` it will use the FRR locks to ensure no events or timers are called in the pthread while the callback is executed. Is that correct ?